plugins/authentication/request-error.js

```diff
@@ -12,11 +12,7 @@ function authenticationRequestError (state, error, options) {
   }
 
   if (error.status === 401 && otpRequired && error.request && error.request.headers['x-github-otp']) {
-    if (state.otp) {
-      delete state.otp // no longer valid, request again
-    } else {
-      throw new HttpError('Invalid one-time password for two-factor authentication', 401, error.headers, options)
-    }
+    throw new HttpError('Invalid one-time password for two-factor authentication', 401, error.headers, options)
```

Comment:
I _think_ we should leave the code above as it was.

When `state.otp` is set, it means that it was valid at least once. If we get a 401 now due to an invalid OTP then we should call `auth.on2fa` again, so we shouldn’t throw an error, only delete the now invalid `state.otp`.

Reply:
You're right, thanks!